File: projects/claude-archunit-automation/test-project/src/main/java/com/example/controller/UserController.java
package com.example.controller;

import com.example.service.UserService;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/users")
public class UserController {

    private final UserService userService;

    public UserController(UserService userService) {
        this.userService = userService;
    }

    @GetMapping("/{id}")
    public String getUser(@PathVariable Long id) {
        return userService.getUser(id);
    }

    @PostMapping
    public String createUser(@RequestBody String user) {
        return userService.createUser(user);
    }
}

File: projects/redis-serializer-poc/string-serializer/src/main/kotlin/com/example/stringser/controller/UserController.kt
package com.example.stringser.controller

import com.example.stringser.domain.User
import com.example.stringser.service.UserCacheService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

data class UserRequest(
    val id: String,
    val name: String,
    val email: String,
    val age: Int,
)

data class RawResponse(
    val serializer: String,
    val key: String,
    val rawValue: String?,
)

@RestController
@RequestMapping("/api/users")
class UserController(
    private val userCacheService: UserCacheService,
) {

    @PostMapping
    fun save(@RequestBody req: UserRequest): ResponseEntity<User> {
        val user = User(id = req.id, name = req.name, email = req.email, age = req.age)
        userCacheService.save(user)
        return ResponseEntity.ok(user)
    }

    @GetMapping("/{id}")
    fun find(@PathVariable id: String): ResponseEntity<User> {
        val user = userCacheService.find(id) ?: return ResponseEntity.notFound().build()
        return ResponseEntity.ok(user)
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: String): ResponseEntity<Unit> {
        userCacheService.delete(id)
        return ResponseEntity.noContent().build()
    }

    @GetMapping("/{id}/raw")
    fun raw(@PathVariable id: String): ResponseEntity<RawResponse> {
        return ResponseEntity.ok(
            RawResponse(
                serializer = "StringRedisSerializer",
                key = "string:user:$id",
                rawValue = userCacheService.raw(id),
            )
        )
    }
}
